Final Project/Server/fiimentor/controllers/authController.ts

```diff
@@ -34,11 +38,124 @@ async function login(req: any, res: any){
     }
 }
 
-function register(req:any, res:any){
+async function register(req: any, res: any) {
     try {
-        //const users = await req.db.User.find({});
 
-       // return res.status(HttpStatus.OK).json();
+        //const user = await req.db.User.find();
+        const serialNumber = req.body.serialNumber;
+        const username = req.body.username;
+        const password = req.body.password;
+        const firstName = req.body.firstName;
+        const lastName = req.body.lastName;
+        const email = req.body.email;
+        let newUser = new User();
+
+
+        let regexpEmail =
```

Comment:
schimba logica, mai intai verifica daca exista in baza de date, abia apoi faci toate verificarile la credentiale